src/electos/ballotmaker/ballots/ballot_layout.py

```diff
@@ -139,42 +138,45 @@ def build_ballot() -> str:
     )
     doc.addPageTemplates(three_column_template)
     doc.addPageTemplates(one_column_template)
-    # add a ballot contest to the second frame (colomn)
-    layout_1 = CandidateContestLayout(
-        CandidateContestData(spacetown_data.can_con_1)
-    )
-    layout_2 = CandidateContestLayout(
-        CandidateContestData(spacetown_data.can_con_2)
-    )
-    layout_3 = CandidateContestLayout(
-        CandidateContestData(spacetown_data.can_con_3)
-    )
-    layout_4 = CandidateContestLayout(
-        CandidateContestData(spacetown_data.can_con_4)
-    )
-    layout_5 = BallotMeasureLayout(
-        BallotMeasureData(spacetown_data.ballot_measure_1)
-    )
-    layout_6 = BallotMeasureLayout(
-        BallotMeasureData(spacetown_data.ballot_measure_2)
-    )
+
+    candidate_contests = []
+    ballot_measures = []
+    # get contests
+    for count, contest in enumerate(ballot_data.contests, start=1):
+        title = contest.title
+        con_type = contest.type
+        logging.info(f"Found contest: {title} - {con_type}")
+        if con_type == CANDIDATE:
+            candidate_contests.append(contest)
+        elif con_type == BALLOT_MEASURE:
+            ballot_measures.append(contest)
+        else:
+            raise ValueError(f"Unknown contest type: {con_type}")
+    logging.info(f"Total: {count} contests.")
+
     elements = []
     # add voting instructions
     inst = Instructions()
-    elements = inst.instruction_list
     elements.append(NextPageTemplate("3col"))
-    elements.append(layout_1.contest_table)
-    elements.append(layout_2.contest_table)
-    elements.append(CondPageBreak(c_height * inch))
-    elements.append(layout_4.contest_table)
-    elements.append(layout_3.contest_table)
+    elements = inst.instruction_list
+
+    # add candidate contests
+    for can_con_count, candidate_contest in enumerate(
+        candidate_contests, start=1
+    ):
+        candidate_layout = CandidateContestLayout(
+            candidate_contest
+        ).contest_table
+        elements.append(candidate_layout)
+        # insert column break after every 2 contests
+        if (can_con_count % 2 == 0) and (can_con_count < 4):
+            elements.append(CondPageBreak(c_height * inch))
+    logging.info(f"Added {can_con_count} candidate contests.")
```

Comment:
Note that this is adding to the _layout_ and not some other list? It's obvious in the code, but would be less obvious in a log.